src/assets/wise5/classroomMonitor/dataExport/strategies/AbstractComponentDataExportStrategy.ts

```diff
@@ -0,0 +1,367 @@
+        extractedUserIDsAndStudentNames['studentName' + (u + 1)] = user.name;
+      }
+    }
+    return extractedUserIDsAndStudentNames;
+  }
+
+  private shouldExportRow(
+    componentState: any,
+    componentStateIndex: number,
+    numComponentStates: number
+  ): boolean {
+    return !(
+      this.includeOnlySubmitsAndIsNotSubmit(componentState) ||
+      this.exportLatestWorkAndIsNotLatestWork(componentStateIndex, numComponentStates)
+    );
+  }
+
+  private includeOnlySubmitsAndIsNotSubmit(componentState: any): boolean {
+    return this.includeOnlySubmits && !componentState.isSubmit;
+  }
+
+  private exportLatestWorkAndIsNotLatestWork(
+    componentStateIndex: number,
+    numComponentStates: number
+  ): boolean {
+    return (
+      this.workSelectionType === 'exportLatestWork' &&
+      componentStateIndex !== numComponentStates - 1
+    );
+  }
+
+  private incrementRevisionCounter(
+    componentRevisionCounter: ComponentRevisionCounter,
+    nodeId: string,
+    componentId: string
+  ): void {
+    componentRevisionCounter.incrementCounter(`${nodeId}_${componentId}`);
+  }
+
+  /**
+   * Create the array that will be used as a row in the student work export
+   * @param rowCounter the current row number
+   * @param workgroupId the workgroup id
+   * @param userId1 the User ID 1
+   * @param userId2 the User ID 2
+   * @param userId3 the User ID 3
+   * @param periodName the period name
+   * @param componentRevisionCounter the mapping of component to revision counter
+   * @param componentState the component state
+   * @return an array containing the cells in the row
+   */
+  createStudentWorkExportRow(
+    rowCounter: number,
+    workgroupId: number,
+    userId1: number,
+    userId2: number,
+    userId3: number,
+    studentName1: string,
+    studentName2: string,
+    studentName3: string,
+    periodName: string,
+    componentRevisionCounter: ComponentRevisionCounter,
+    componentState: any
+  ): string[] {
+    const row = new Array(this.columnNames.length);
+    row.fill('');
+    row[this.columnNameToNumber.get('#')] = rowCounter;
+    row[this.columnNameToNumber.get('Workgroup ID')] = workgroupId;
+    this.setStudentIDsAndNames(
+      row,
+      userId1,
+      studentName1,
+      userId2,
+      studentName2,
+      userId3,
+      studentName3
+    );
+    row[this.columnNameToNumber.get('Class Period')] = periodName;
+    row[this.columnNameToNumber.get('Project ID')] = this.configService.getProjectId();
+    row[this.columnNameToNumber.get('Project Name')] = this.projectService.getProjectTitle();
+    row[this.columnNameToNumber.get('Run ID')] = this.configService.getRunId();
+    row[this.columnNameToNumber.get('Student Work ID')] = componentState.id;
+    const formattedDateTime = this.utilService.convertMillisecondsToFormattedDateTime(
+      componentState.serverSaveTime
+    );
+    row[this.columnNameToNumber.get('Server Timestamp')] = formattedDateTime;
```

Comment:
No need to store in variable since we don't use it again?
```suggestion
    row[this.columnNameToNumber.get('Server Timestamp')] = this.utilService.convertMillisecondsToFormattedDateTime(
      componentState.serverSaveTime
    );
```